agent/agentssh/agentsshrsa/key.go

```diff
@@ -0,0 +1,85 @@
+package agentsshrsa
+
+import (
+	"crypto/rsa"
+	"math/big"
+	"math/rand"
+)
+
+// DeterministicPrivateKey generates an RSA private key deterministically based on the provided seed.
+// This function uses a deterministic random source to generate the primes p and q, ensuring that the
+// same seed will always produce the same private key. The generated key is 2048 bits in size.
+func DeterministicPrivateKey(seed int64) *rsa.PrivateKey {
```

Comment:
Might also be good to provide a link to the reference implementation for comparison.

Reply:
You mean as in the implementation of the std lib or reference in the sense of RSA algorithm reference?

Reply:
The stdlib implementation.